internal/pubsub/rabbitpubsub/rabbit.go

```diff
@@ -304,7 +281,11 @@ func (t *topic) As(i interface{}) bool {
 	if !ok {
 		return false
 	}
-	*c = t.conn
+	conn, ok := t.conn.(*connection)
+	if !ok { // running against the fake; pretend to succeed
```

Comment:
This seems weird... why not just return `false`? Maybe the test verification part should check to see if we're running against the fake instead...?